Last-Update: 2016-02-15
Forwarded: no
Bug-Upstream: https://github.com/ariya/phantomjs/issues/13727
Author: Ximin Luo <infinity0@debian.org>
Reviewed-By: Dmitry Smirnov <onlyjob@debian.org>
Description: Port to Qt 5.5
 - In webpage, don't inherit QWebFrame::PrintCallback since it's not public

--- a/src/webpage.cpp
+++ b/src/webpage.cpp
@@ -1254,9 +1254,9 @@
     }
 
     printer.setPageMargins(marginLeft, marginTop, marginRight, marginBottom, QPrinter::Point);
 
-    m_mainFrame->print(&printer, this);
+    m_mainFrame->print(&printer);
     return true;
 }
 
 void WebPage::setZoomFactor(qreal zoom)
--- a/src/webpage.h
+++ b/src/webpage.h
@@ -44,9 +44,9 @@
 class NetworkAccessManager;
 class QWebInspector;
 class Phantom;
 
-class WebPage : public QObject, public QWebFrame::PrintCallback
+class WebPage : public QObject
 {
     Q_OBJECT
     Q_PROPERTY(QString title READ title)
     Q_PROPERTY(QString frameTitle READ frameTitle)
